src/main/java/org/hmhb/program/Program.java

```diff
@@ -49,6 +53,20 @@
     @NotNull
     private String coordinates;
 
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(
```

Comment:
This tells JPA (hibernate) which columns out of which tables so it can join the appropriate tables when filling in the Program's List<County>.  You can see the SQL it generates in the console output and logs (because I've turned on the show_sql property in application.properties).